### ISSUE TITLE
Windows 11 24H2 as host - Configuration 12.0 VMs aren't using NVidia drivers (Error 43)

### ISSUE THREAD

**Issue by dante102** [jamesstringerparsec/Easy-GPU-PV#407]
### Description
Virtual machines created using Hyper-V on a "Windows 11 24H2" host cannot utilize the gpu after executing the GPU-PV script.
Inside the VM, The "*dxdiag*" command does not show the "*Render*" tab, and the "*Display*" tab shows an error message: 
![изображение](https://github.com/user-attachments/assets/8ef90e46-fdad-4b6e-9fc2-afd42f145863)

### Steps to reproduce
**Method 1: New VM on 24H2 Host**
1. Host OS: "Windows 11 24H2".
2. Create a Gen2 VM with "Windows 11 24H2" as the guest OS.
3. Run the "Easy-GPU-PV" script (see solution #404 if driver file copying fails).
4. Launch the VM, enter **Run** (Win+R)" and execute "*dxdiag*".

**Method 2: Existing VM Conversion (the long way)**
1. Host OS: "Windows 11 22H2".
2. Create a Gen2 VM with Windows 11 22H2.
3. Run the "Easy-GPU-PV" script (see solution #404 if driver file copying fails).
3.1. **Optional:** Launch the VM, open **Run** (Win+R), and execute "*dxdiag*" to verify functionality - should show the "Render" tab and no errors in "*Display*" tab.
4. Close the VM and update the Host to 24H2.
4.1. **Optional:**  Launch the VM, re-run "*dxdiag*" to confirm functionality - should show the "Render" tab and no errors in "*Display*" tab.
5. Recreate a VM using the virtual hard disk of the existing Gen2 VM.
6. Launch the new VM, open **Run** (Win+R)", and re-run "dxdiag".

### Expected Behavior
Inside the VM, the "Render" tab should appear in dxdiag, indicating that the GPU is functional.
### Actual Behavior
The "*Render*" tab is missing, and the GPU is not functional.

**Comment by timminator** [comment]
Cannot replicate. Works fine on my end. Are you using the latest Nvidia driver? I am on 565.90.

**Comment by dante102** [comment]
Correct, i am using the latest display driver (version 565.90). 

I will be closing this thread. The issue was found and it was me not being able to follow basic instructions - i have been launching the wrong script (Interactive-Easy-GPU-PV instead of Easy-GPU-PV).

The Easy-GPU-PV had no issues finding the drive letter and the "Render" tab is visible in "dxdiag".

**Comment by e-t-l** [comment]
@dante102 Could you explain a little bit more about which script you were accidentally using, and which one worked? Do you mean you were running scripts from the [Razzmatazzz/Interactive-Easy-GPU-PV](https://github.com/Razzmatazzz/Interactive-Easy-GPU-PV) repo? What specific script from Easy-GPU-PV did you run that worked? Did you have to change any parameters?

I'm having an identical issue (https://github.com/jamesstringerparsec/Easy-GPU-PV/issues/424) after updating my graphics drivers for the first time. I ran Update-VMGpuPartitionDriver.ps1 but dxdiag still isn't recognizing the updated drivers...

**Comment by e-t-l** [comment]
Wait you're saying the Dynamic RAM setting is what caused your driver problems?

**Comment by e-t-l** [comment]
Dang it. I tried disabling Dynamic RAM and it still didnt work. Could I trouble you to share a screenshot of your current (working) dxdiag in the VM?

**Comment by dante102** [comment]
Here's how the dxdiag currently looks on my VM:
![Image](https://github.com/user-attachments/assets/73d5bcfa-4515-4dff-805e-253a882b75f1)
![Image](https://github.com/user-attachments/assets/896c1243-b627-41c7-8a8a-8c7ff20c9a54)

**Comment by e-t-l** [comment]
So in the second image, it still says it's using Windows' "vrd.sys" drivers with version 10.0.26100.1150, which is totally different than up-to-date Nvidia drivers used by the host OS (according to DxDiag run on the host OS). But you're saying your GPU is properly recognized? 

Your dxdiag looks the same as mine, so maybe I'm not having the exact same issue.. When Steam does acknowledge my GPU, it throws an error saying, "your graphics drivers are extremely out of date," which makes me think it's only recognizing those June 2006 Windows drivers...

**Comment by LewisWildgoose** [comment]
> So in the second image, it still says it's using Windows' "vrd.sys" drivers with version 10.0.26100.1150, which is totally different than up-to-date Nvidia drivers used by the host OS (according to DxDiag run on the host OS). But you're saying your GPU is properly recognized?
> 
> Your dxdiag looks the same as mine, so maybe I'm not having the exact same issue.. When Steam does acknowledge my GPU, it throws an error saying, "your graphics drivers are extremely out of date," which makes me think it's only recognizing those June 2006 Windows drivers...

Did you manage to find a fix for this? I'm having the exact same issue whereby my Host OS shows driver version 32.x or whatever in Device Manager, but every time I create a VM it installs Windows/Microsoft Drivers version 10 under my 3060 in Device Manager. Driving me up the wall trying to fix it.

**Comment by timminator** [comment]
@LewisWildgoose That it reports a Microsoft driver in the VM is perfectly fine and expected. As long as their is no exclamation mark or warning next to it in the device manager it should be fine.
The question is how your performance is. Have you run a game or something like that?

**Comment by LewisWildgoose** [comment]
> [@LewisWildgoose](https://github.com/LewisWildgoose) That it reports a Microsoft driver in the VM is perfectly fine and expected. As long as there is no exclamation mark or warning next to it in the device manager it should be fine. The question is how your performance is. Have you run a game or something like that?

Thanks for the reply mate! And I've tried to run a game but it errors out unless I run it in compatibility mode, and also if I try to open Nvidia Control Panel I get an error saying that no GPU is detected. Device Manager does show the GPU with no issues thankfully, it's just when trying to run games it crashes on start. I actually used to use this months ago and didn't have the same issue, but have upgraded Windows etc since.

**Comment by timminator** [comment]
That the Nvidia control panel isn't working in the VM is also expected and normal behavior.
But when the same game ran a few months ago, then something is indeed wrong.
Have you tried something like Unigine Heaven? That works definitely.

**Comment by LewisWildgoose** [comment]
Ah ok perfect thanks for confirmation! I played around a lot more, with updates, restarts, redeploying the VM etc and finally gotten it working! I have no idea what actually fixed it which is annoying, but it's working and that's the main bit for me, thanks for the help!

**Comment by timminator** [comment]
Are these the error messages you mentioned in #424? You said Steam reported that there is no GPU detected, but its actually the game in these screenshots...
If that were all you issues you had the whole time, then you never really had a problem... :-)
GPU Paravirtualization is a "hack" by Microsoft and not officially supported by Nvidia or AMD so error messages like these can definitely occur. When the game runs, everything is fine.
You can also open a terminal in your VM and there you can run the command `nvidia-smi`. If everyting is correctly setup, this will return the correct driver-version, CUDA version, GPU name and a few other infos.

**Comment by e-t-l** [comment]
> Are these the error messages you mentioned in #424? You said Steam reported that there is no GPU detected, but its actually the game in these screenshots...

Yes, talking about the same error, and you're right it is technically the game's exe, not steam.exe. I wasn't thinking about them as separate things but I guess I should have.

> You can also open a terminal in your VM and there you can run the command nvidia-smi. If everyting is correctly setup, this will return the correct driver-version, CUDA version, GPU name and a few other infos.

Thank you, this is a lifesaver! In nvidia-smi I can see that it does report the right driver version at the top.

> so error messages like these can definitely occur. When the game runs, everything is fine.

I guess it's all fine then. The main thing that had me worried was that I'm almost positive that I did NOT get those error messages when I was on driver version 566.36, and it only started after updating to 572.16. That's why I thought the VM didn't update correctly. (I hadn't yet tested rolling back the drivers to v566.36 on both host machine and VM, as that seems like potentially more troublesome than it's worth.)

But maybe the error message is just some weird quirk with this game executable, like you said, and it doesn't mean anything since the game does seem to run just fine. (I feel like I have also been getting more occasional freezing frames and stuttering lately, but that might also just be because I'm out of the tutorial zone and the game actually has more stuff to render now, lol). Thank you for all this feedback, it has been very reassuring. 

P.S. @timminator I'm a big fan of [Moonlight-Tailored-for-GPU-PV](https://github.com/timminator/Moonlight-Tailored-for-GPU-PV)! Do you plan to release newer versions of it to stay up to date with upstream Moonlight?

**Comment by timminator** [comment]
> P.S. @timminator I'm a big fan of [Moonlight-Tailored-for-GPU-PV](https://github.com/timminator/Moonlight-Tailored-for-GPU-PV)! Do you plan to release newer versions of it to stay up to date with upstream Moonlight?

@e-t-l We informed the maintainers about the connection issue when using Sunshine/Moonlight in VMs with GPU-Paravirtualization in June.
They were able to provide a fix for that in October.
In January they finally released a new offical version with this fix in place and shortly afterwards another one. In the first one you will also see the fix in the [Changelog](https://github.com/LizardByte/Sunshine/releases/tag/v2025.118.1518408).
With this fix my modified version is no longer necessary. You just need to update Sunshine in your VM. Afterwards you can switch back to the official version of the Moonlight client.

I also plan to update my Enhanced-GPU-PV project, so that it installs the newest Sunshine version out of the box. I also plan to update the Readme instructions with the latest news accordingly, but I had not the time for that until now.
I also plan to add some other small convenient features - I found for example a way that we no longer have a black screen when a UAC prompt appears.

**Comment by ThinkPuc** [comment]
> The 36% percent was my fan speed at that time, thats completely irrelevant. If you can run the command "nvidia-smi" successfully you are good - thats the main point.

but i cant use parsec , i have error say -14003, is there a way to fix
